tests/ratchet/test_utils_ratchet.c

```diff
@@ -0,0 +1,496 @@
+        TEST_ASSERT(false);
+    }
+
+    return memcmp(sender_chain1->private_key, sender_chain2->private_key, sizeof(sender_chain1->private_key)) == 0 &&
+           memcmp(sender_chain1->public_key, sender_chain2->public_key, sizeof(sender_chain1->public_key)) == 0 &&
+           ratchet_chain_key_cmp(&sender_chain1->chain_key, &sender_chain2->chain_key);
+}
+
+static bool
+ratchet_cmp(vscr_ratchet_t *ratchet1, vscr_ratchet_t *ratchet2) {
+
+    return memcmp(ratchet1->root_key, ratchet2->root_key, sizeof(ratchet1->root_key)) == 0 &&
+           ratchet_sender_chain_cmp(ratchet1->sender_chain, ratchet2->sender_chain) &&
+           ratchet1->prev_sender_chain_count == ratchet2->prev_sender_chain_count &&
+           ratchet_receiver_chains_cmp(ratchet1->receiver_chains, ratchet2->receiver_chains) &&
+           ratchet_skipped_msgs_cmp(ratchet1->skipped_messages, ratchet2->skipped_messages);
+}
+
+static bool
+ratchet_session_cmp(vscr_ratchet_session_t *ratchet_session1, vscr_ratchet_session_t *ratchet_session2) {
+
+    return memcmp(ratchet_session1->sender_identity_public_key, ratchet_session2->sender_identity_public_key,
+                   sizeof(ratchet_session1->sender_identity_public_key)) == 0 &&
+           memcmp(ratchet_session1->sender_ephemeral_public_key, ratchet_session2->sender_ephemeral_public_key,
+                   sizeof(ratchet_session1->sender_ephemeral_public_key)) == 0 &&
+           memcmp(ratchet_session1->receiver_long_term_public_key, ratchet_session2->receiver_long_term_public_key,
+                   sizeof(ratchet_session1->receiver_long_term_public_key)) == 0 &&
+           memcmp(ratchet_session1->receiver_one_time_public_key, ratchet_session2->receiver_one_time_public_key,
+                   sizeof(ratchet_session1->receiver_one_time_public_key)) == 0 &&
+           ratchet_session1->is_initiator == ratchet_session2->is_initiator &&
+           ratchet_session1->received_first_response == ratchet_session2->received_first_response &&
+           ratchet_session1->receiver_has_one_time_public_key == ratchet_session2->receiver_has_one_time_public_key &&
+           ratchet_cmp(ratchet_session1->ratchet, ratchet_session2->ratchet);
+}
+
+void
+restore_session(vscr_ratchet_session_t **session) {
+    vscr_error_ctx_t error_ctx;
+    vscr_error_ctx_reset(&error_ctx);
+
+    vscr_ratchet_session_t *session_ref = *session;
+
+    vsc_buffer_t *buffer = vsc_buffer_new_with_capacity(vscr_ratchet_session_serialize_len(*session));
+
+    vscr_ratchet_session_serialize(*session, buffer);
+
+    *session = vscr_ratchet_session_deserialize(vsc_buffer_data(buffer), &error_ctx);
+
+    vsc_buffer_destroy(&buffer);
+
+    TEST_ASSERT_EQUAL(vscr_SUCCESS, error_ctx.error);
+
+    vscr_ratchet_session_setup_defaults(*session);
+
+    bool flag = ratchet_session_cmp(session_ref, *session);
+
+    if (!flag) {
+        TEST_ASSERT(false);
+    }
+
+    vscr_ratchet_session_destroy(&session_ref);
+}
+
+#endif
```

Comment:
Code style: ensures that there is always at lease 1 empty line at the end of the file.